### PR TITLE
Action form

### PR DESCRIPTION
**Pull request by zaucker**
Needed for CrdMgr CM-138:
- New base class `AbstractAction`
- Derive `AbstractForm` from it
- Add action-only frontend class `ActionForm`

### CODE REVIEW

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+ */
+qx.Class.define("callbackery.ui.plugin.ActionForm", {
+    extend : qx.ui.container.Composite,
+    /**
+     * create a page for the View Tab with the given title
+     *
+     * @param cfg {Object} plugin configuration map
+     * @param getFormData {Function} method to get form data
+     */
+    construct : function(cfg,getFormData) {
```
Comment (oetiker) [inline lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js:20]: what is the use case for 'getFormData' ?

Comment (zaucker) [reply lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js]: Ah, dumm benannt. Es ist parentFormData ... umbenannt.

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+                    else {
+                        if (btCfg.addToToolBar !== false) {
+                            if (btCfg.action == 'cancel') {
+                                this._btnRow.add(button);
+                            }
+                            else {
+                                this._grid.setRowAlign(row, 'left', 'middle');
+                                this._form.add(button, {row : row, column : btCfg.btnColumn || colBtn});
+                                if (btCfg.description) {
+                                    description = new qx.ui.basic.Label(this.xtr(btCfg.description || ''));
```
Comment (oetiker) [inline lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js:278]: this only happens if there is a description ... so I guess the `|| ''` is not necessary

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+        _form: null,
+        _grid : null,
+        _btnRow : null,
+        _tableMenu: null,
+        _defaultAction: null,
+        _buttonMap: null,
+
+        _populate: function(cfg,buttonClass,getParentFormData){
+            var menues = {};
+            let row = 0, colDesc= 1, colBtn = 0;
```
Comment (oetiker) [inline lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js:116]: is it `colDesc` or  `descColumn` ...

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        getButtonSetMap: function(){
+            return this._buttonSetMap;
+        }
+    },
+    destruct : function() {
+        if (! this._buttonMap) {
+            return;
+        }
+        for (const [key, btn] of Object.entries(this._buttonMap)) {
+            console.log('destroying', btn);
```
Comment (oetiker) [inline lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js:308]: no logging

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js (hunk 2 above)
Comment (zaucker) [reply lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js:278]: Removed

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js (hunk 3 above)
Comment (zaucker) [reply lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js:116]: Renamed both `colDesc` abd `btnDesc`.

File: lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js (hunk 4 above)
Comment (zaucker) [reply lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js:308]: Removed.

**Pull request merged**